infra/src/main/kotlin/fi/fta/geoviite/infra/common/KmNumber.kt

```diff
@@ -53,9 +53,13 @@ data class KmNumber @JsonCreator(mode = DISABLED) constructor(
 private const val METERS_MAX_INTEGER_DIGITS = 4
 private const val METERS_MAX_DECIMAL_DIGITS = 6
 private fun limitScale(meters: BigDecimal) =
-    if (meters.scale() < 0) meters.setScale(0)
-    else if (meters.scale() > METERS_MAX_DECIMAL_DIGITS) meters.setScale(METERS_MAX_DECIMAL_DIGITS, HALF_UP)
-    else meters
+    if (meters.scale() < 0) {
```

Comment:
Detekt nillitti tästä niin lisäsin sulkuja. En nyt oikeastaan tätä välttämättä parempana muotona pidä, mutta kai se vähän lisäturvaa tuo 🤷

Reply:
Joo nämä on näitä preferenssijuduja ja mihin on tottunut. Itse taas ihmettelen kun iffittelyehdon jälkeen onkin suoraan ilman blokkia jo operaatio (ja huhhuh niitä missä se elsekin on samalla rivillä)

Tässä sen sijaan voisi käyttää when-blokkia, niin nuo saisi samalle riville halutessaan. Whenissä tietysti myös se hyöty, että se varmistaa kaikkien casejen käsittelyn.

```
    when {
        (meters.scale() < 0) -> meters.setScale(0)
        (meters.scale() > METERS_MAX_DECIMAL_DIGITS) -> meters.setScale(METERS_MAX_DECIMAL_DIGITS, HALF_UP)
        
        else -> meters
    }

    // Tai ilmankin sulkuja
    when {
        meters.scale() < 0 -> meters.setScale(0)
        meters.scale() > METERS_MAX_DECIMAL_DIGITS -> meters.setScale(METERS_MAX_DECIMAL_DIGITS, HALF_UP)
        
        else -> meters
    }

Reply:
:+1: whenille, se on kyllä hyvä näihin.

Njoo, if ja else samalla rivillä on vähän inhottava lukea, mutta se on myös ainoa tapa tehdä kotlinissa ternääri, joten olen vain hyväksynyt sen jos kyse on niin lyhyestä että mahtuu yhteen riviin :shrug: